src/main/java/org/terasology/flexiblepathfinding/PathfinderSystem.java

```diff
@@ -82,20 +72,30 @@ public int requestPath(EntityRef requester, Vector3i target, List<Vector3i> star
         return requestPath(config, callback);
     }
 
+    private void processPath(PathfinderCallback callback, JPSConfig config) {
+        JPSImpl jps = new JPSImpl(config);
+        List<Vector3i> path = Lists.newArrayList();
+        try {
+            if (jps.run()) {
+                path = jps.getPath();
+            }
+        } catch (InterruptedException e) {
+            logger.error("Process Interrupted", e);
+        }
+        this.completePathFor(config.requester);
+        callback.pathReady(path, new Vector3i()); // unusued :??
+    }
+
     public int requestPath(JPSConfig config, PathfinderCallback callback) {
         if (config.requester != null && config.requester.exists()) {
             if (entitiesWithPendingTasks.contains(config.requester)) {
                 return -1;
             }
             entitiesWithPendingTasks.add(config.requester);
         }
-
-        if (config.executor == null) {
-            config.executor = workerTaskMaster.getExecutorService();
```

Comment:
The question I have left on this review is: If this assignment of `config.executor` goes away, is there anything left using `config.executor` at all?

and if not, does that mean we never use TimeLimiter?

Reply:
Probably that's the case (although we should double-check), yes.
Question for me is, if we don't use it, do we create a negative impact on performance? And if so, how can we model the timeout logic using Reactor / `GameScheduler`?